vscode-lean4/src/infoview.ts

```diff
@@ -123,6 +153,25 @@ export class InfoProvider implements Disposable {
                 this.client.convertRange(show.selection)
             );
         },
+
+        createRpcSession: async uri => {
+            const sessionId = await rpcConnect(this.client, uri);
+            const session = new RpcSession(this.client, sessionId, uri);
```

Comment:
I have a dumb question, do we need to do any keepAlives?  The RpcNeedsReconnect handling seems to work really well, so why not just let work after a 30 second delay in typing.  That way my battery life is increased even more... ?

Reply:
If I am understanding the question correctly, the reason why we need to keep the session alive rather than just reconnect every now and then is that reconnects clear the serverside storage of RPC-referenced objects thus invalidating parts of the UI, for example the state of a running widget.

Reply:
#73 is an example of a problem we have because the reconnect doesn't work well.